Management/DecisionEngine.cs

```diff
@@ -34,17 +42,56 @@ public class DecisionEngine : IDecisionEngine
             _covidDataClient = covidDataClient ?? throw new ArgumentNullException(nameof(covidDataClient));
             _weatherDataClient = weatherDataClient ?? throw new ArgumentNullException(nameof(covidDataClient));
             _airQualityDataClient = airQualityDataClient ?? throw new ArgumentNullException(nameof(airQualityDataClient));
+            _clients = new List<IWeightedClient>
+            {
+                _covidDataClient,
+                _weatherDataClient,
+                _airQualityDataClient
+            };
         }
 
         /// <summary>
         /// Calculate the desired location using weighted scores from COVID-19, weather, and air quality.
         /// </summary>
         /// <returns>The weighted score.</returns>
-        public Task<IEnumerable<Recommendation>> CalculateDesiredLocationAsync()
+        public async Task<Dictionary<City, double>> CalculateDesiredLocationAsync(
+            State state,
+            CountryCode countryCode,
+            CancellationToken cancellationToken)
         {
-            var result = (0.3 * _covidDataClient.CalculateScoreAsync()) + (0.3 * _weatherDataClient.CalculateScoreAsync()) + (0.3 * _airQualityDataClient.CalculateScoreAsync());
+            var cityBag = new ConcurrentBag<IEnumerable<City>>();
+
+            var cityTasks = _clients.Select(async client =>
+            {
+                var result = await client.GetDefaultCitiesAsync(state, countryCode, cancellationToken);
+                cityBag.Add(result);
+            });
+
+            await Task.WhenAll(cityTasks);
+
+            var commonCities = new HashSet<City>(cityBag.SelectMany(bag => bag.Select(city => city))).Take(10);
+
+            var airQualityDataClientResult = await _airQualityDataClient.CalculateScoresAsync(commonCities, state, countryCode, cancellationToken);
```

Comment:
Why I dont use parllel execution here?
Because as I mentioned in previous comment, the result will not be in order therefore there is no way we can tell which is from which client. 
We need to obtain this info in order to apply the weighted index.